apps/backend/src/modules/contributors/contributor.repository.ts

```diff
@@ -135,50 +93,29 @@ class ContributorRepository implements Repository {
 			query.whereILike("contributors.name", `%${contributorName}%`);
 		}
 
-		const contributorsWithProjectsAndEmails = await query.execute();
-
-		return {
-			items: contributorsWithProjectsAndEmails.map((contributor) => {
-				return ContributorEntity.initialize(contributor);
-			}),
-		};
-	}
-
-	public async findAllWithoutPagination({
-		contributorName,
-		hasHidden = true,
-	}: {
-		contributorName?: string;
-		hasHidden?: boolean;
-	}): Promise<{ items: ContributorEntity[] }> {
-		const query = this.contributorModel
-			.query()
-			.select("contributors.*")
-			.select(
-				raw(
-					"COALESCE(ARRAY_AGG(DISTINCT jsonb_build_object('id', projects.id, 'name', projects.name)) FILTER (WHERE projects.id IS NOT NULL), '{}') AS projects",
-				),
-			)
-			.leftJoin("git_emails", "contributors.id", "git_emails.contributor_id")
-			.leftJoin("activity_logs", "git_emails.id", "activity_logs.git_email_id")
-			.leftJoin("projects", "activity_logs.project_id", "projects.id")
-			.groupBy("contributors.id")
-			.withGraphFetched("gitEmails");
-
-		if (!hasHidden) {
-			query.whereNull("contributors.hiddenAt");
+		if (projectId) {
+			query.havingRaw("?? = ANY(ARRAY_AGG(projects.id))", projectId);
```

Comment:
Is there a reason we can't just use `where` here?

Reply:
No we can't. Check [this](https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/401#discussion_r1774008896) comment, where I've described the reason

Reply:
Thanks for always providing the links :cat2: